src/AutoMapper/MapperConfiguration.cs

```diff
@@ -244,6 +254,25 @@ public bool AllowNullCollections
                     //});
                 }
 
+                var genericMapperType = FindClosedGenericObjectMapperFor(types);
+
+                if (genericMapperType != null)
+                {
+                    var srcParam = Parameter(typeof(TSource), "src");
+                    var destParam = Parameter(typeof(TDestination), "dest");
+                    var ctxtParam = Parameter(typeof(ResolutionContext), "ctxt");
+
+                    var mapExpression = Lambda(
+                        ToType(Call(
+                            New(genericMapperType),

```

Comment:
One other motivation is I'd like to see about encouraging these over type
converters. Those are a bit strange because none of the other type map
configuration applies.

Heck, if we can get the config API right we might be able get rid of type
converters all together. It's already weird to have it all mixed together.

On Wednesday, April 27, 2016, Tyler Carlson <notifications@github.com>
wrote:

> In src/AutoMapper/MapperConfiguration.cs
> <https://github.com/AutoMapper/AutoMapper/pull/1250#discussion_r61355424>:
>
> > @@ -244,6 +254,25 @@ public bool AllowNullCollections
> >                      //});
> >                  }
> >
> > +                var genericMapperType = FindClosedGenericObjectMapperFor(types);
> > +
> > +                if (genericMapperType != null)
> > +                {
> > +                    var srcParam = Parameter(typeof(TSource), "src");
> > +                    var destParam = Parameter(typeof(TDestination), "dest");
> > +                    var ctxtParam = Parameter(typeof(ResolutionContext), "ctxt");
> > +
> > +                    var mapExpression = Lambda(
> > +                        ToType(Call(
> > +                            New(genericMapperType),
>
> I still hate you have to new it up every time. I would much rather it be
> static call. Yes you lose interfaces, but at least save on newing up an
> object on every map.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/AutoMapper/AutoMapper/pull/1250/files/3f7b7d1cdd261cad06891677b91bf625090473d9#r61355424>
>